local/local.go

```diff
@@ -43,58 +38,53 @@ func UpdateBuildAgent() error {
 	return nil
 }
 
-// nolint: gocyclo
-// TODO(zzak): This function is fairly complex, we should refactor it
-func Execute(opts BuildOptions) error {
-	for _, f := range opts.Args {
-		if f == "--help" || f == "-h" {
-			return opts.Help()
-		}
+func Execute(flags *pflag.FlagSet, cfg *settings.Config) error {
+
+	processedArgs, configPath := extractConfigPathFromFlags(flags)
+	cl := client.NewClient(cfg.Host, cfg.Endpoint, cfg.Token, cfg.Debug)
+	configResponse, err := api.ConfigQuery(cl, configPath)
+
+	if err != nil {
+		return err
+	}
+
+	if !configResponse.Valid {
+		return fmt.Errorf("config errors %v", configResponse.Errors)
 	}
 
-	if err := validateConfigVersion(opts.Args); err != nil {
+	processedConfigPath, err := writeStringToTempFile(configResponse.OutputYaml)
```

Comment:
Could you add a `defer os.Remove(processedConfigPath)` after this, or does the file need to last longer than this function?

Reply:
yeah, the file needs to persist until after the docker command has executed, but `syscall.Exec` replaces the current process, so a `defer` would never be called. I'll add a comment to note that a file will be left in `/tmp` after each build.